jupiterone/data_source_user_group.go

```diff
@@ -0,0 +1,169 @@
+package jupiterone
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/jupiterone/terraform-provider-jupiterone/jupiterone/internal/client"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &userGroupDataSource{}
+	_ datasource.DataSourceWithConfigure = &userGroupDataSource{}
+)
+
+// NewUserGroupDataSource is a helper function to simplify the provider implementation.
+func NewUserGroupDataSource() datasource.DataSource {
+	return &userGroupDataSource{}
+}
+
+// userGroupDataSource is the data source implementation.
+type userGroupDataSource struct {
+	version string
+	qlient  graphql.Client
+}
+
+// Metadata implements resource.Resource
+func (*userGroupDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_user_group"
+}
+
+// Schema implements resource.Resource
+func (*userGroupDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "A saved JupiterOne User Group",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "The name of the user group",
+			},
+			"description": schema.StringAttribute{
+				Computed:    true,
+				Description: "The description of the user group",
+			},
+			"permissions": schema.ListAttribute{
+				Description: "A list of permissions for the user group.",
+				Computed:    true,
+				ElementType: types.StringType,
+			},
+			"query_policy": schema.ListAttribute{
+				Description: "A list of query policy statements for the user group.",
+				Computed:    true,
+				ElementType: types.MapType{
+					ElemType: types.ListType{
+						ElemType: types.StringType,
+					},
+				},
+			},
+		},
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (d *userGroupDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data UserGroupModel
+
+	// Read Terraform state into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	groups, err := client.GetGroupsByName(ctx, d.qlient, data.Name.ValueString())
```

Comment:
Why get groups by name instead of getting a group by id?

Reply:
Why _did_ I do that? There must have been some reason. Going to think and then probably just convert to get by id...

Reply:
OH! I know why here. I am allowing the consumer to just pass the name to retrieve the resource.  This is different than the resource_user_group, where I do grab the resource by id.

This data_source is used like so:
```
data "jupiterone_user_group" "administrators" {
  name = "Administrators"
}
```